src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/DebugPropertyPage/RemoveStartArgumentsActionHandler.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Xml;
+using Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.Designer;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties;
+
+[Export(typeof(ILinkActionHandler))]
+[ExportMetadata("CommandName", "RemoveStartArguments")]
+internal class RemoveStartArgumentsActionHandler : ILinkActionHandler
+{
+    private const string PropertyNameToDelete = "StartArguments";
+    private const string UserSuffix = ".user";
+    private string? _xmlFile;
+
+    public Task HandleAsync(UnconfiguredProject project, IReadOnlyDictionary<string, string> editorMetadata)
+    {
+        _xmlFile = project.FullPath + UserSuffix;
+
+        _ = project.Services.ThreadingPolicy.JoinableTaskFactory.RunAsync(RemoveElementAsync);
```

Comment:
This does not fix the threading issue.

The previous code awaited the work's completion, and did all that work on the UI thread.

This code will still do that work on the UI thread, only it won't await the completion. This means that the user could make changes that have a race condition with the below logic. For example, they could double-click the hyperlink, in which case you might run `RemoveElementAsync` twice.

Instead, I think you should await `RemoveElementAsync` as before, but inside that method switch to a background thread via `await TaskScheduler.Default`.

You shouldn't need `JTF.RunAsync` here as there's no work to join. The caller will have done that for you anyway.

Reply:
fixed with TaskScheduler.Default